src/Bicep.Core/Utils/Environment.cs

```diff
@@ -6,4 +6,7 @@ public class Environment : IEnvironment
 {
     public string? GetVariable(string variable)
         => System.Environment.GetEnvironmentVariable(variable);
+
+    public IEnumerable<string> GetVariableNames()
+        => System.Environment.GetEnvironmentVariables().Keys.Cast<string>();
```

Comment:
Very nitpicky: the keys of `System.Environment.GetEnvironmentVariables()`'s return type *should* always be strings, but if you use `.OfType<string>()` instead of `.Cast<string>()`, the enumeration will skip any non-string env variables rather than throwing a cast exception. I would probably opt for the former but can see the argument for preferring the latter.